### ISSUE TITLE
Delete old artifacts after a successfull action

### ISSUE THREAD

**Issue by JoseQuintas** [actions/toolkit#699]
Imagine this situation:

An automatic workflow to create artifacts after each commit: 300MB of artifact
20 commits on each day x 90 days x 300MB = 540.000MB = 540GB
How to delete old runs of the same artifact after successfull creation?
This will reduce from 540GB to 300MB, and do not need to wait for 90 days.

At momment, may be I can create scripts to search what to delete.
Is there exists actions to delete specific workflow run or a list of workflow runs
or old runs of same workflow, when current workflow is sucessfull ?
From 540GB to 300MB, there exists a big difference, to user and to github.

Another option is to have repo artifacts, like release or pack.
Each workflow could replace the repo artifact, and not the workflow artifact.
Think about to reduce from 540GB to 300MB.

And we are talking about ONE workflow, think about all repositories on github, and how many space this can reduce at all.

**Comment by jdrydn** [comment]
Similar discussion happening here: https://github.com/actions/upload-artifact/issues/5

**Comment by jdrydn** [comment]
@JoseQuintas Have you considered setting [the retention period](https://github.com/actions/upload-artifact#retention-period)?

**Comment by JoseQuintas** [comment]
Yes, but on this way, if no commit no artifact.

**Comment by thboop** [comment]
Hey @JoseQuintas, we have customizable retention periods per artifact. You can set that at a repo or organization level, or in the [job](https://github.com/actions/upload-artifact#retention-period)

Can you describe what you want in more detail beyond that? We strongly recommend using the retention settings as a way to cleanup.

> if no commit no artifact.

I'm not sure what you mean by this, could you rephrase?

I'm going to close this out for now, but we can reopen once we understand the feature request better!
